cmd/juno/juno.go

```diff
@@ -202,8 +205,19 @@ func NewCmd(config *node.Config, run func(*cobra.Command, []string) error) *cobr
 
 		// TextUnmarshallerHookFunc allows us to unmarshal values that satisfy the
 		// encoding.TextUnmarshaller interface (see the LogLevel type for an example).
-		return v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
-			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc())))
+		if err := v.Unmarshal(config, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(
+			mapstructure.TextUnmarshallerHookFunc(), mapstructure.StringToTimeDurationHookFunc()))); err != nil {
+			return err
+		}
+
+		if v.GetString(customNetworkF) != "" {
+			err := config.Network.SetCustomNetwork(v.GetString(customNetworkF))
+			if err != nil {
+				return err
+			}
```

Comment:
I'm starting to doubt my idea about using a json string on the command line. Can we change it to a file name?
Sorry for the back-and-forth on this part.